service/tracker/consensus_test.go

```diff
@@ -0,0 +1,908 @@
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		require.NoError(t, db.Update(operation.IndexBlockHeight(header.Height, header.ID())))
+		require.NoError(t, db.Update(operation.IndexPayloadSeals(header.ID(), sealIDs)))
+		for _, seal := range seals {
+			require.NoError(t, db.Update(operation.InsertSeal(seal.ID(), seal)))
+		}
+
+		cons := tracker.BaselineConsensus(t, tracker.WithDB(db), tracker.WithLast(header.Height))
+
+		_, err := cons.Seals(header.Height + 999)
+
+		assert.Error(t, err)
+	})
+
+	t.Run("handles missing block height index in DB", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		// Missing block height index in database.
+		//require.NoError(t, db.Update(operation.IndexBlockHeight(header.Height, header.ID())))
+		require.NoError(t, db.Update(operation.IndexPayloadSeals(header.ID(), sealIDs)))
+		for _, seal := range seals {
+			require.NoError(t, db.Update(operation.InsertSeal(seal.ID(), seal)))
+		}
+
+		cons := tracker.BaselineConsensus(t, tracker.WithDB(db), tracker.WithLast(header.Height))
+
+		_, err := cons.Seals(header.Height)
+
+		assert.Error(t, err)
+	})
+
+	t.Run("handles missing seal index in DB", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		require.NoError(t, db.Update(operation.IndexBlockHeight(header.Height, header.ID())))
+		// Missing seal index in database.
+		//require.NoError(t, db.Update(operation.IndexPayloadSeals(header.ID(), sealIDs)))
+		for _, seal := range seals {
+			require.NoError(t, db.Update(operation.InsertSeal(seal.ID(), seal)))
+		}
+
+		cons := tracker.BaselineConsensus(t, tracker.WithDB(db), tracker.WithLast(header.Height))
+
+		_, err := cons.Seals(header.Height)
+
+		assert.Error(t, err)
+	})
+
+	t.Run("handles missing seals in DB", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		require.NoError(t, db.Update(operation.IndexBlockHeight(header.Height, header.ID())))
+		require.NoError(t, db.Update(operation.IndexPayloadSeals(header.ID(), sealIDs)))
+		// Missing seal index in database.
+		//for _, seal := range seals {
+		//	require.NoError(t, db.Update(operation.InsertSeal(seal.ID(), seal)))
+		//}
+
+		cons := tracker.BaselineConsensus(t, tracker.WithDB(db), tracker.WithLast(header.Height), tracker.WithLast(header.Height))
+
+		_, err := cons.Seals(header.Height)
+
+		assert.Error(t, err)
+	})
+}
+
+func TestConsensus_Commit(t *testing.T) {
+	header := mocks.GenericHeader
+	blockData := mocks.GenericBlockData()
+
+	t.Run("nominal case", func(t *testing.T) {
+		t.Parallel()
+
+		db := helpers.InMemoryDB(t)
+		defer db.Close()
+
+		require.NoError(t, db.Update(operation.IndexBlockHeight(header.Height, header.ID())))
+
+		holder := mocks.BaselineRecordHolder(t)
+		holder.RecordFunc = func(blockID flow.Identifier) (*uploader.BlockData, error) {
+			assert.Equal(t, header.ID(), blockID)
+
```

Comment:
I don't get which lines you refer to by

> We have three lines at different levels and scopes of the function without space

For me it's about separating assertions and return statements. They're at the same level yes, but for me the only case where I'd cuddle a `return` statement is the following:

```go
func x() *x {
	result := x{}
	return &x
}
```

And even then, only if the variable definition is a one-liner.

I'd be fine with not doing that but then maybe we need some kind of style guide/ruleset because I'm very much used to my old style guide (which was mostly inspired by https://github.com/uber-go/guide/blob/master/style.md).

Reply:
```go
// on the same line, even though different scope and/or indentation
holder := mocks.BaselineRecordHolder(t)
holder.RecordFunc = func(blockID flow.Identifier) (*uploader.BlockData, error) {
	assert.Equal(t, header.ID(), blockID)
```

```go
	assert.Equal(t, header.ID(), blockID)
	// space here, even though same scope and indentation level
	return record, nil
```

Spaces are, in my view, used to put statements into blocks that belong together as much as possible. That's also why I don't like when you don't put a space after an opening brace for the function body. Here, you keep statements together that are a lot less related than the statements to pull apart. Doesn't seem logical to me.